### PR TITLE
Migrate 'components/announcement_bar/default_announcement_bar' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by morganrconnolly**
#### Summary

Updates `components/announcement_bar/default_announcement_bar/` to typsescript
#### Ticket Link
fixes https://github.com/mattermost/mattermost-server/issues/16119

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

**Comment by hanzei** [comment]
Hey @morganrconnolly,

Thanks for raising this PR :+1: CI complains about a couple of linter issues. Could you please take a look? You can find the log [here](https://app.circleci.com/pipelines/github/mattermost/mattermost-webapp/28968/workflows/be9d834d-415e-4da3-9b4c-a62aac094d17/jobs/191203/parallel-runs/0/steps/0-103).

**Comment by morganrconnolly** [comment]
@hanzei I fixed the linter errors. Thanks for letting me know what needed to happen next! C:

**Comment by hanzei** [comment]
Thanks @morganrconnolly! Could you please run `npm run updatesnapshot` to let the tests pass?

**Comment by morganrconnolly** [comment]
@hanzei done C:

**Comment by hanzei** [comment]
@morganrconnolly Awesome! Queued your PR for review

File: components/announcement_bar/default_announcement_bar/announcement_bar.tsx [hunk 1]
```diff
@@ -17,39 +16,37 @@ import ToggleModalButtonRedux from 'components/toggle_modal_button_redux';
...
         textColor: '',
         type: AnnouncementBarTypes.CRITICAL,
-        handleClose: null,
-        onButtonClick: null,
         showLinkAsButton: false,
         isTallBanner: false,
     }
 
-    componentDidMount() {
+    public componentDidMount() {// public for testing
```
Comment (hahmadia) [inline components/announcement_bar/default_announcement_bar/announcement_bar.tsx:49]:
```suggestion
    // "public" due to this function being used in the test file
    public componentDidMount() {
```
I'd make the comment above the function and make it more clear

File: package-lock.json [hunk 2]
```diff
@@ -6995,7 +6995,7 @@
         },
         "p-cancelable": {
           "version": "0.4.1",
-          "resolved": "https://registry.npmjs.org/p-cancelable/-/p-cancelable-0.4.1.tgz",
+          "resolved": "http://registry.npmjs.org/p-cancelable/-/p-cancelable-0.4.1.tgz",
```
Comment (hahmadia) [inline package-lock.json:6998]: Please revert the changes in this file. We don't need to be changing anything in the packagelock.json file.
Thanks!

**Review by hahmadia — CHANGES_REQUESTED** [review]
Some minor suggestions. Thanks!

File: components/announcement_bar/default_announcement_bar/announcement_bar.tsx (hunk 1 above)
Comment (morganrconnolly) [reply components/announcement_bar/default_announcement_bar/announcement_bar.tsx:49]: These methods actually did not need to be public.

**Comment by morganrconnolly** [comment]
@hanzei incorporated your feedback !

**Review by hahmadia — APPROVED** [review]
Thanks LGTM!

**Review by nevyangelova — APPROVED** [review]
LGTM!

**Comment by morganrconnolly** [comment]
sweet. @nevyangelova @hahmadia let me know if there's anything else I can do here!

**Review by saturninoabril — APPROVED** [review]
Thanks @morganrconnolly for your contribution! LGTM, tested and passed.
